### PR TITLE
Exercise Mode Editing Capabilities

### PR DESCRIPTION
**Pull request by russell-rozenbaum**
**TL;DR**
Exercise Mode is fully (mostly) editable

**Added Features**
Adds ability to edit all* Exercise Mode features:
- Title
- Module name
- Prompt
- Mutation Tests
- Implementation Tests
*Still need to add editing for Syntax Tests

Also adds BlankExercise.ml for a completely blank exercise spec. Might be useful when creating new exercise button... just make sure to change UUID.

**Not Bugs, but Relatively Important Things for Editing an Exercise**

- [x] Cannot edit mutation test hints
- [ ] Cannot add/remove Syntax Validation tests
- [x] Cannot edit Syntax Validation test hints
- [x] Cannot edit Implementation Grading test hints

### CODE REVIEW

**Comment by facundoy** [comment]
Found the following visual bug: 

- [x] Confirmation and Cancel buttons are not displaying properly in editing mode for Implementation Grading and Test Validation

**Review by cyrus- — COMMENTED** [review]
- Instructor Mode should be off by default when you load an exercise, since it now includes a bunch of things we don't want new users to have to think about.
- Prompt should not be bolded by default.
- Cut/copy/paste does not work in exercise mode editors.
- Not sure how hard this is, but unicode symbols are mangled when included in the prompt, e.g. I wanted to use unicode triple equals ≡ but it does not work.
- Is it possible for the hints to be a field under each mutant, rather than separately in the Mutation Testing box?
- Syntax Validation box should include a way to edit the % discount on points.
- Instead of "New point max" just say "Points"
- Too much space between exercise title, module name and prompt.

**Comment by facundoy** [comment]
- [x] Instructor Mode should be off by default when you load an exercise, since it now includes a bunch of things we don't want new users to have to think about.
- [x] Prompt should not be bolded by default.
- [x] Cut/copy/paste does not work in exercise mode editors.
- [x] Unicode for prompt
- [ ] Is it possible for the hints to be a field under each mutant, rather than separately in the Mutation Testing box?
- (Removing this task. The current architecture for exercise mode can't support this (or would be a much more involved project to implement).) Syntax Validation box should include a way to edit the % discount on points.
- [x] Instead of "New point max" just say "Points"
- [x] Too much space between exercise title, module name and prompt.

**Comment by facundoy** [comment]
Unicode seems to be working on prompt locally in this format &#x2261;

edit: meant that unicode works when inputing the following "&#x2261"

**Comment by cyrus-** [comment]
@facundoy regarding Unicode, this is what happens when I enter 
image

It turns into an a in this case:
image

When I edit it again I get an unknown character:
image

**Comment by russell-rozenbaum** [comment]
A few bugs:
- [ ] Unable to refresh Hazel after opening exercise mode. This at least happens locally for me. My assumption is a problem with how the exercises are stored.  

Still need to implement:
- [ ]  Hints to be a field under each mutant, rather than separately in the Mutation Testing box? This should just involve restructuring the HTML layout / CSS

**Comment by cyrus-** [comment]
@russell-rozenbaum @facundoy there are a few conflicts to resolve here. could we get this PR fixed up and ready to merge this week?

**Comment by russell-rozenbaum** [comment]
@facundoy There is a bug where the page freezes whenever I try to confirm edits in the prompt. Does this happen for you as well? Any idea what this could be?

**Comment by facundoy** [comment]
@russell-rozenbaum is your last commit solving this bug? I tested it locally with that change and the prompt editing seems to be working without issue, and it was working fine when I solved the merge conflicts this weekend if I remember correctly. 

> @facundoy There is a bug where the page freezes whenever I try to confirm edits in the prompt. Does this happen for you as well? Any idea what this could be?

**Comment by russell-rozenbaum** [comment]
> @russell-rozenbaum is your last commit solving this bug? I tested it locally with that change and the prompt editing seems to be working without issue, and it was working fine when I solved the merge conflicts this weekend if I remember correctly.
> 
> > @facundoy There is a bug where the page freezes whenever I try to confirm edits in the prompt. Does this happen for you as well? Any idea what this could be?

@facundoy No, my most recent commit only fixed a bug with reloading a page while editing. This seems to be a different issue. Whenever I make an edit in the prompt edit box specifically, the page becomes unresponsive. I don't think this is only on my machine, as it happens on both my local server and the branch's server.

**Comment by cyrus-** [comment]
@russell-rozenbaum or @facundoy Russell let me know that this never actually got merged. there is a conflict in 1 file right now, can one of you resolve that and we can get this merged?

**Comment by russell-rozenbaum** [comment]
Well, I'm not sure what [cdaba5e](https://github.com/hazelgrove/hazel/pull/1444/commits/cdaba5eac830fa6540bf8104e384d35754461421) did exactly, but [e34bbcd](https://github.com/hazelgrove/hazel/pull/1444/commits/e34bbcdf4c2516fd77ae54d939c99638e5f8f89c) was where I finalized things...  cdaba5e should be fine and ready to be merged
